### PR TITLE
Adds default_placeholder_pattern config option

### LINKED ISSUE #297: Default Placeholder Pattern config
[Request change (Wiki)]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues/new?template=change_request_wiki.md
[Bug report]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues/new?template=bug_report.md
[issues]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues
[wiki]: https://github.com/PlaceholderAPI/PlaceholderAPI/wiki
[Pull Requests]: https://github.com/PlaceholderAPI/PlaceholderAPI/pulls

## Please read
This template is only for requesting new functions to be added to PlaceholderAPI or for old one to be changed!  
Please use the [Request change (Wiki)] template to ask for additions/changes to the wiki.  
For reporting bugs of PlaceholderAPI use the [Bug report] template.

Also make sure that the [wiki] doesn't already mention such a function/change and that there isn't a issue about this change on the [issues] page.

We recommend to [open a Pull Request][Pull Requests] for making changes to PlaceholderAPI's code.

- [ ] New function for PlaceholderAPI.  
A new function that developers could use.
- [x] Change to code (Internal).  
Changes to code that won't affect the end-user.
- [ ] Change to code (External).  
Changes to code that will affect the end-user (breaks stuff).
- [ ] Other: __________ 

The default placeholder pattern (``[%]([^%]+)[%]``) is inherently flawed for the purpose of nesting/ recursively resolving placeholder:

``%placeholder1_%placeholder2%%`` will resolve ``%placeholder1_%`` first, leaving ``placeholder2%%`` behind.

The bracket placeholder pattern and relational placeholder pattern solve this issue. However, most plugin developers will likely choose to use the following functions for resolving placeholders (and retrieving other info), since they are worded as if they were the default:
```
setPlaceholders(OfflinePlayer player, List<String> text)
setPlaceholders(OfflinePlayer player, List<String> text, boolean colorize)
setPlaceholders(OfflinePlayer player, String text)
setPlaceholders(OfflinePlayer player, String text, boolean colorize)
setPlaceholders(Player player, String text)
setPlaceholders(Player player, String text, boolean colorize)
setPlaceholders(Player player, List<String> text)
setPlaceholders(Player player, List<String> text, boolean colorize)
containsPlaceholders(String text)
getPlaceholderPattern()
```

I suggest adding a new config option ``default-placeholder-pattern``. This option could either be a choice between the three current patterns (should add a 4th pattern that is formatted like the relational pattern so they won't interfere e.g.: ``[%](ph_)([^%]+)[%]``) or a String where you can specify a placeholder pattern yourself. This option would be set to the the current default pattern by default.

The functions mentioned above would then be copied to new functions named something like ``setPercentSignPlaceholders`` and the old functions would be changed such that they use the pattern set in the config.

I would also suggest renaming ``PLACEHOLDER_PATTERN`` to ``PERCENT_SIGN_PLACEHOLDER_PATTERN`` for clarity. ``PLACEHOLDER_PATTERN`` may be used as field name of the placeholder pattern set via config.

Since the config will have ``[%]([^%]+)[%]`` set as the default pattern this change will not break backwards compatibility.

If this suggestion gets accepted I would be happy to make a PR containing the changes discussed.

### PR DESCRIPTION
**Pull request by CreepyCre**
[Pull requests]: https://github.com/PlaceholderAPI/PlaceholderAPI/pulls
[contributing file]: https://github.com/PlaceholderAPI/PlaceholderAPI/tree/master/.github/CONTRIBUTING.md

## Please read
Please make sure you checked the following:
- You checked the [Pull requests] page for any upcoming changes.
- You documented any public code that the end-user might use.
- You followed the [contributing file] 

### Type

- [x] Internal change (Doesn't affect end-user).
- [ ] External change (Does affect end-user).
- [ ] Other: __________ 

closes #297

Adds default_placeholder_pattern in config. This pattern will now be used in functions that previously used PLACEHOLDER_PATTERN. By default it is set to the same pattern as PLACEHOLDER_PATTERN to maintain backwards compatibility.

PLACEHOLDER_PATTERN has been renamed to PERCENT_SIGN_PLACEHOLDER_PATTERN, the functions using it previously were recreated with appropriate names.

### CODE REVIEW

**Comment by Andre601** [comment]
You should give PR titles a proper, descriptive name and use the `closes #297` somewhere in the description.

File: src/main/java/me/clip/placeholderapi/PlaceholderAPI.java [hunk 1]
```diff
@@ -477,12 +487,21 @@ public static boolean unregisterExpansion(PlaceholderExpansion ex) {
...
-  
+
+  /**
+   * Gets the placeholder pattern for the default placeholders.
+   *
+   * @return The pattern for {@literal %<identifier>_<params>%}
+   */
+  public static Pattern getPercentSignPlaceholderPattern() {
+    return PERCENT_SIGN_PLACEHOLDER_PATTERN;
+  }
```
Comment (Andre601) [inline src/main/java/me/clip/placeholderapi/PlaceholderAPI.java:503]: There is `getPlaceholderPattern()` which returns the `%key_value%` Pattern.

Comment (CreepyCre) [reply src/main/java/me/clip/placeholderapi/PlaceholderAPI.java:503]: ``getPlaceholderPattern()`` has been changed to now return the default_placeholder_pattern set in the config.

**Comment by Andre601** [comment]
@CreepyCre Is this still up to date? The methods and general code have recently changed a lot, so you would need to update your code here to make it mergable again (See the above conflics)

**Comment by Sxtanna** [comment]
This is no longer something that will be pursued by the API.

**Pull request closed without merging**